### PR TITLE
#1-4Task. Ihar Tsykala

### PR DESCRIPTION
**Pull request by IharTsykala**
https://travis-ci.org/github/IharTsykala/js-assignments/builds/711449400

### CODE REVIEW

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -130,7 +130,9 @@ function repeatString(value, count) {
  *   'ABABAB','BA' => 'ABAB'
  */
 function removeFirstOccurrences(str, value) {
-    throw new Error('Not implemented');
+  const index = str.indexOf(value);
+	if (index === -1) return str;
+	return str.slice(0, index) + str.slice(index + value.length);
```
Comment (KostyaKov) [inline task/01-strings-tasks.js]: A simpler option is to use the method "replace()"

File: task/01-strings-tasks.js [hunk 2]
```diff
@@ -221,7 +226,21 @@ function getRectangleString(width, height) {
  *
  */
 function encodeToRot13(str) {
-    throw new Error('Not implemented');
+  const inputBase = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
+  const outBase =   'NOPQRSTUVWXYZABCDEFGHIJKLMnopqrstuvwxyzabcdefghijklm';
+  let str2 = '';
+    for(let i = 0; i<str.length; i++) {
```
Comment (KostyaKov) [inline task/01-strings-tasks.js]: It is necessary to follow the generally accepted code style: 
indentation after "for", "if", as well as indentation around operators "i <n".

File: task/01-strings-tasks.js [hunk 3]
```diff
@@ -238,36 +257,42 @@ function encodeToRot13(str) {
  *   isString(new String('test')) => true
  */
 function isString(value) {
-    throw new Error('Not implemented');
+    return {}.toString.call(value) === "[object String]"
```
Comment (KostyaKov) [inline task/01-strings-tasks.js]: A better option for such a task is to use the "typeof" operator

File: task/01-strings-tasks.js [hunk 4]
```diff
@@ -238,36 +257,42 @@ function encodeToRot13(str) {
...
  *     ...
  *   'Q♠' => 50
  *   'K♠' => 51
  */
 function getCardId(value) {
-    throw new Error('Not implemented');
+ const arr = ['A♣','2♣','3♣','4♣','5♣','6♣','7♣','8♣','9♣','10♣','J♣','Q♣','K♣',
+            'A♦','2♦','3♦','4♦','5♦','6♦','7♦','8♦','9♦','10♦','J♦','Q♦','K♦',
+            'A♥','2♥','3♥','4♥','5♥','6♥','7♥','8♥','9♥','10♥','J♥','Q♥','K♥',
+            'A♠','2♠','3♠','4♠','5♠','6♠','7♠','8♠','9♠','10♠','J♠','Q♠','K♠']
```
Comment (KostyaKov) [inline task/01-strings-tasks.js]: You should put the symbol ";" at the end of the const declaration.
It is also necessary to observe indentation "for" and spaces after "for", and between the operators "<".
https://javascript.info/coding-style

File: task/02-numbers-tasks.js [hunk 5]
```diff
@@ -54,7 +52,15 @@ function getCicleCircumference(radius) {
  *  -3, 3  => 0
  */
 function getAverage(value1, value2) {
-    throw new Error('Not implemented');
+  let x = 0
```
Comment (KostyaKov) [inline task/02-numbers-tasks.js]: You need to use a simpler solution to find the average of two numbers.

File: task/03-date-tasks.js [hunk 6]
```diff
@@ -76,14 +77,29 @@ function isLeapYear(date) {
...
-   throw new Error('Not implemented');
+  let duration = endDate-startDate
+  let hours = Math.floor(duration / (3600*1000))
+  if(hours<10) hours = '0' + hours
+  let minutes = Math.floor((duration - (hours*3600*1000))/(60*1000))
+  if(minutes<10) minutes = '0' + minutes
+  let seconds = Math.floor((duration - (hours*3600*1000)-(minutes*60*1000))/1000)
+  if(seconds<10) seconds = '0' + seconds
+  let milliseconds = duration - (hours*3600*1000) - (minutes*60*1000) - (seconds*1000)
+  if( milliseconds<100)  milliseconds = '0' +  milliseconds
```
Comment (KostyaKov) [inline task/03-date-tasks.js]: It is necessary to follow the general style of the code: 
- indentation after the keywords "for", "if", etc.
- spaces between operators "*", "<", etc.
- remove extra spaces "return`  $ {hours}: " and "if ( milliseconds", and so on.
Also, comments that are not related to the current code are not needed.

**Review by KostyaKov — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by IharTsykala** [comment]
Please, accept my pr

File: .eslintrc.js [hunk 7]
```diff
@@ -1,23 +1,23 @@
...
+  },
+  rules: {
+    "require-yield": "off",
+    "no-unused-vars": "off",
+    "no-prototype-building": "off",
+    "no-var": "error",
+    "no-lonely-if": "error",
+    "no-proto": "error",
+    "no-else-return": "error",
+    // "eqeqeq": "error",
```
Comment (ilya4321) [inline .eslintrc.js]: Please revert .eslintrc.js to initial state

**Comment by IlyaMokin** [comment]
@IharTsykala Please check and fix comments above.
I've also found that the PR has redundant changes in styling, spaces, comments and redundant tasks (except 1-4)
Please follow the simple rules in the future:
 - do not add any non logic changes to code if review is required
 - if the changes in styling are important - you can do it as separate PR with styling only changes to avoid noise in the main PR

**Comment by IharTsykala** [comment]
https://travis-ci.org/github/IharTsykala/js-assignments/builds/712874065

**Comment by IharTsykala** [comment]
> @IharTsykala Please check and fix comments above.
> I've also found that the PR has redundant changes in styling, spaces, comments and redundant tasks (except 1-4)
> Please follow the simple rules in the future:
> 
> * do not add any non logic changes to code if review is required
> * if the changes in styling are important - you can do it as separate PR with styling only changes to avoid noise in the main PR

I have global setting lint when i save something in my VSCode. Must I changes another PR(in task js-assignments) or was your note  for next tasks. If it was for next tasks, not problem, I will be writing code in webStrorm when isnt a global lint.

**Comment by IlyaMokin** [comment]
It was for that PR, you need to exclude changes which is not related to the task1-4. 
This is part of the task. One of the goal is get a skill to make readable PR. With current approach in real project that will be impossible to make some code review because of redundant in noise.

PS: the issues are not related to VSCode or WebStrorm - you can use any IDE

**Comment by IharTsykala** [comment]
https://travis-ci.org/github/IharTsykala/js-assignments/builds/713225533

I excluded changes in exist code and comment. Please, accept my pr

**Comment by IlyaMokin** [comment]
Hm.. you can see changed files and comments [here](https://github.com/AisBrestEDU/js-assignments/pull/785/files). I still see changes in eslint and so on... also a few unresolved comments

**Comment by IharTsykala** [comment]
I resolve comments and replace code eslintrc.js as first view. Please, accept my pr

https://travis-ci.org/github/IharTsykala/js-assignments/builds/713762774

**Comment by IlyaMokin** [comment]
"You need to exclude changes which is not related to the task1-4"
I see the same issue here and in taks6. 
You edited comments, that created redundant noise in PRs. In real project that makes PRs unreadable and has no chance to be reviewed as a rule. 
image

**Comment by IharTsykala** [comment]
https://travis-ci.org/github/IharTsykala/js-assignments/builds/714650256

I exclude changes. Please, accept my pr

**Comment by IharTsykala** [comment]
https://travis-ci.org/github/IharTsykala/js-assignments/builds/719248882

I exclude changes. Please, accept my pr

**Comment by vramaniuk** [comment]
Closing the PR as accepted! Thanks!

**Comment by IlyaMokin** [comment]
The PR is accepted
